File: SKA-api/src/main/java/com/example/skaapi/Controllers/AlumnoController.java
package com.example.skaapi.Controllers;
import com.example.skaapi.Models.Alumno.Alumno;
import com.example.skaapi.Services.AlumnoService;
import io.swagger.v3.oas.annotations.parameters.RequestBody;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Optional;

@RestController
@RequestMapping("/api")
public class AlumnoController {

    AlumnoService alumnoService;

    AlumnoController(AlumnoService alumnoService){ this.alumnoService = alumnoService;
    }

    @GetMapping("alumnos")
    List<Alumno> getAlumnos(){
        return alumnoService.getAll();
    }

    @GetMapping("alumnos/{id}")
    Optional<Alumno> getAlumno(@PathVariable String id){
        return alumnoService.getAlumnoPerMatricula(id);
    }

    @PostMapping("alumnos")
    Alumno postAlumnos(@RequestBody Alumno alumno){
        return alumnoService.insert(alumno);
    }
}

File: SKA-Backend/bin/main/com/churris/SKABackend/controllers/AlumnoController.kt
package com.churris.SKABackend.controllers

import com.churris.SKABackend.models.Test
import com.churris.SKABackend.services.DatabaseService
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.ui.set
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api")
class MessageResource(val service: DatabaseService) {
    @GetMapping
    fun getAllPlayers() = service.getAll()

    @GetMapping("/{id}")
    fun getPlayer(@PathVariable id: Long) = service.getById(id)

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun savePlayer(@RequestBody test: Test): Test? = service.create(test)

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deletePlayer(@PathVariable id: Long) = service.remove(id)

    @PutMapping("/{id}")
    fun updatePlayer(
        @PathVariable id: Long, @RequestBody test: Test
    ) = service.update(id, test)
}